src/main/java/spoon/support/adaption/GlueNode.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * SPDX-License-Identifier: (MIT OR CECILL-C)
+ *
+ * Copyright (C) 2006-2019 INRIA and contributors
+ *
+ * Spoon is available either under the terms of the MIT License (see LICENSE-MIT.txt) of the Cecill-C License (see LICENSE-CECILL-C.txt). You as the user are entitled to choose the terms under which to adopt Spoon.
+ */
+package spoon.support.adaption;
+
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.CtTypeParameter;
+import spoon.reflect.reference.CtTypeParameterReference;
+import spoon.reflect.reference.CtTypeReference;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+import static spoon.support.adaption.NodePrintHelper.quote;
+import static spoon.support.adaption.NodePrintHelper.toJsonLikeArray;
+
+class GlueNode {
+
+	private final List<CtTypeReference<?>> actualArguments;
+	private final CtTypeReference<?> inducedBy;
+	private final Set<DeclarationNode> lowerNodes;
```

Comment:
Same comment as on `DeclarationNode`

```suggestion
	private final List<DeclarationNode> children;
```